dotCMS/src/main/java/com/dotcms/saml/DotSamlConfigurationServiceImpl.java

```diff
@@ -0,0 +1,231 @@
+package com.dotcms.saml;
+
+import com.dotcms.saml.service.external.IdentityProviderConfiguration;
+import com.dotcms.saml.service.external.SamlConfigurationService;
+import com.dotcms.saml.service.external.SamlException;
+import com.dotcms.saml.service.external.SamlName;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This service will retrieve information form the idp config, but also if the value is not set will provide the default value for the saml name.
+ * It has different implementation for several kinds of values
+ * @author jsanca
+ */
+public class DotSamlConfigurationServiceImpl implements SamlConfigurationService {
+
+    private static final String UNABLE_TO_READ_FILE = "File does not exist or unable to read : ";
+    private static final String NOT_FOUND_ERROR = "Property Name not Found: ";
+    /**
+     * To set into the init map context, the absolute path of the default properties for SAML.
+     */
+    public final static String DOT_SAML_DEFAULT_PROPERTIES_CONTEXT_MAP_KEY = "dotSamlDefaultPropertiesContextMapKey";
+
+    private AtomicBoolean init = new AtomicBoolean(false);
```

Comment:
Codacy found an issue: [Private field 'init' could be made final; it is only initialized in the declaration or constructor.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)